libs/compact-map/test/Test/Compact/SplitMap.hs

```diff
@@ -0,0 +1,238 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Test.Compact.SplitMap where
+
+import Data.Compact.KeyMap (Key (..))
+import Data.Compact.SplitMap
+import qualified Data.Compact.SplitMap as SplitMap
+import Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+import Data.Set (Set)
+import Data.Word (Word64)
+import Test.Compact.KeyMap
+import Test.Tasty
+import Test.Tasty.QuickCheck
+import Prelude hiding (lookup)
+
+-- ==============================================================
+-- All the features of TxIn, but we don't need to import it
+
+newtype MockTxIn = MockTxIn (Int, Key)
+  deriving (Show, Eq, Ord)
+
+instance Split MockTxIn where
+  split (MockTxIn pair) = pair
+  join n k = MockTxIn (n, k)
+
+instance Arbitrary MockTxIn where
+  arbitrary = join <$> chooseInt (0, 25) <*> arbitrary
```

Comment:
Not so sure about that.  The SS instance has non-uniformly distributed keys, with small numbers in the first component, this will test paths in the KeyMap code, that won't be tested using the  MockTxIn instance

Reply:
Yeah, I understand that, but that is a specific property of `KeyMap`, which isn't particularly interesting for `SplitMap` tests, right? In any case, my point was that there is no need to create `MockTxIn`, a regular `(Int, Key)` tuple will do just fine. Feel free to keep `SS` as is if you think there is some benefit in it